src/main/java/com/rackspace/salus/telemetry/ambassador/web/controller/AgentHistoryController.java

```diff
@@ -50,13 +53,15 @@ public AgentHistoryController(AgentHistoryService agentHistoryService)  {
       Pageable pageable)
       throws NotFoundException {
     if(!StringUtils.isEmpty(envoyId) && !StringUtils.isEmpty(resourceId)) {
-      throw new IllegalArgumentException("EnvoyId and ResourceId both cannot be not-null");
+      throw new IllegalArgumentException("envoyId and resourceId both cannot be set");
     } else if(!StringUtils.isEmpty(envoyId))  {
-      return PagedContent.ofSingleton(
-          new AgentHistoryDTO(agentHistoryService.getAgentHistoryForTenantAndEnvoyId(tenantId, envoyId)
-          .orElseThrow(() ->
-              new NotFoundException(
-                  String.format("No Agent History found for tenant %s and envoy %s ", tenantId, envoyId)))));
+      Optional<AgentHistory> optional = agentHistoryService.getAgentHistoryForTenantAndEnvoyId(tenantId, envoyId);
+      if(optional.isPresent())  {
+        return PagedContent.ofSingleton(
+            new AgentHistoryDTO(optional.get()));
+      } else  {
+        return PagedContent.ofSingleton(null);
```

Comment:
Hmm, I guess we need a `PageContent.empty()` method since `ofSingleton(null)` will produce an awkward response that looks like an element count of 1 but the one element will be null.

For now can just do
```suggestion
        return PagedContent.fromPage(Page.empty());
```